Increase test coverage for GammuDispatcher

lunkwill42 opened the pull request:
Playing with using CodeCov for the NAV repo, I picked a module with low test coverage, with the goal of increasing its test coverage. I picked GammuDispatcher, the smsd dispatcher plugin that can dispatch SMS messages using the Gammu library.

This PR:
* Rewrites the existing tests to become pure pytest tests.
* Adds more tests for the actual GammuDispatcher plugin code
* Removes the GammuDispatcher's obsolete dependency on `six`, since NAV code is now Python 3-only.

The tests were designed to run even when then gammu library isn't present, but since the mock library's patch framework doesn't work well when the code to patch doesn't exist, the test module needs to mock the bare minimum of the gammu library if it isn't present.

hmpf commented on tests/unittests/smsd/gammudispatcher_test.py:
```diff
@@ -1,31 +1,91 @@
+            with pytest.raises(PermanentDispatcherError):
+                dispatcher.sendsms("999999", ["Message 1", "Message 2"])
+
+    def test_should_raise_permanent_error_when_gammu_errors_during_init(
+        self, mock_gsm_error
+    ):
+        mocked_statemachine = Mock(Init=Mock(side_effect=mock_gsm_error))
+        with patch('gammu.StateMachine') as statemachine:
+            statemachine.return_value = mocked_statemachine
+            dispatcher = GammuDispatcher(None)
+            with pytest.raises(PermanentDispatcherError):
+                dispatcher.sendsms("999999", ["Message 1", "Message 2"])
+
+    def test_should_raise_permanent_error_when_gammu_errors_during_init(
+        self, mock_gsm_error
+    ):
+        mocked_statemachine = Mock(Init=Mock(side_effect=mock_gsm_error))
+        with patch('gammu.StateMachine') as statemachine:
+            statemachine.return_value = mocked_statemachine
+            dispatcher = GammuDispatcher(None)
+            with pytest.raises(PermanentDispatcherError):
+                dispatcher.sendsms("999999", ["Message 1", "Message 2"])
```
Duplicate methods?

lunkwill42 replied:
yes. gone now

hmpf reviewed (commented):
Looks like there's a duplicate test-method.

lunkwill42 commented:
> Looks like there's a duplicate test-method.

Indeed, I probably didn't read through very well before pushing my existing code :)

hmpf commented:
I looked at the sonarcloud-report: It's reporting duplication in the mibs, but those are all really data and should be excepted.

lunkwill42 commented:
> I looked at the sonarcloud-report: It's reporting duplication in the mibs (`TABLE`s), but those are all really data and should be excepted.

Yes, I've added SonarCloud to the workflow now, and added inclusion/exception config for the service. This should ensure bundled libraries and generated files/data are excluded from analysis.

Only annoying thing at the moment is that SonarCloud assumes all SQL is PL/SQL (i.e. Oracle), and complains about errors that do not affect PostgreSQL.
